### PR TITLE
tests: help flag

### PR DESCRIPTION
**Pull request by evenstensberg**
**What kind of change does this PR introduce?**
Improving tests

**Did you add tests for your changes?**
Yaaa

**If relevant, did you update the documentation?**
N/A
**Summary**
Tests the help flag, improves some of the source for outputting the help section by doing an early return.

**Does this PR introduce a breaking change?**
No

**Other information**

Also adds code-owners file

### CODE REVIEW

File: lib/bootstrap.js [hunk 1]
```diff
@@ -4,11 +4,34 @@ const cmdArgs = require('command-line-args');
 
 require('./utils/process-log');
 
+const helpFlagPresent = (args) => {
+    return args.find(arg => arg === 'help' || arg === '--help');
```
Comment (jamesgeorge007) [inline lib/bootstrap.js]:
```suggestion

```

File: lib/bootstrap.js [hunk 2]
```diff
@@ -4,11 +4,34 @@ const cmdArgs = require('command-line-args');
 
 require('./utils/process-log');
 
+const helpFlagPresent = (args) => {
```
Comment (jamesgeorge007) [inline lib/bootstrap.js]:
```suggestion
const helpFlagPresent = (args) => return args.find(arg => ['help', '--help'].indexOf(arg) !== -1);
```

File: lib/bootstrap.js [hunk 3]
```diff
@@ -4,11 +4,34 @@ const cmdArgs = require('command-line-args');
...
+    return args.find(arg => arg === 'help' || arg === '--help');
+}
```
Comment (jamesgeorge007) [inline lib/bootstrap.js]:
```suggestion

```

File: lib/bootstrap.js [hunk 4]
```diff
@@ -4,11 +4,34 @@ const cmdArgs = require('command-line-args');
...
+
+const normalizeFlags = (args, command) => {
+    return args.slice(2).filter(arg => arg.indexOf('--') < 0 && arg !== command.name && arg !== command.alias);
+}
```
Comment (jamesgeorge007) [inline lib/bootstrap.js]: a similar approach would do well here.

File: lib/bootstrap.js [hunk 5]
```diff
@@ -4,11 +4,34 @@ const cmdArgs = require('command-line-args');
...
+
+const isCommandUsed = (commands) => {
+    return commands.find(cmd => {
+        if (cmd.alias) {
+            return process.argv.includes(cmd.name) || process.argv.includes(cmd.alias);
+        }
```
Comment (jamesgeorge007) [inline lib/bootstrap.js]: Minor tweaks possible :thinking:

**Comment by jamesgeorge007** [comment]
@evenstensberg left a couple of comments.

File: test/binCases/help/mixed/mixed.test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+'use strict';
```
Comment (ematipico) [inline test/binCases/help/mixed/mixed.test.js]: All the tests are about the help command. Why did you decide to create a separate file for each case instead to have one file with different cases there?

File: lib/bootstrap.js [hunk 7]
```diff
@@ -4,11 +4,23 @@ const cmdArgs = require('command-line-args');
 
 require('./utils/process-log');
 
+const helpFlagPresent = args => args.find(arg => ['help', '--help'].indexOf(arg) !== -1);
+
+const normalizeFlags = (args, command) => args.slice(2).filter(arg => arg.indexOf('--') < 0 && arg !== command.name && arg !== command.alias);
+
+const isCommandUsed = commands => commands.find(cmd => (cmd.alias ? process.argv.includes(cmd.name) || process.argv.includes(cmd.alias) : process.argv.includes(cmd.name)));
```
Comment (ematipico) [inline lib/bootstrap.js]: The previous version was more readable. Would you mind keep that?

File: lib/bootstrap.js [hunk 8]
```diff
@@ -4,11 +4,23 @@ const cmdArgs = require('command-line-args');
 
 require('./utils/process-log');
 
+const helpFlagPresent = args => args.find(arg => ['help', '--help'].indexOf(arg) !== -1);
```
Comment (ematipico) [inline lib/bootstrap.js]: `isHelpFlagPresent`

File: lib/bootstrap.js (hunk 7 above)
Comment (jamesgeorge007) [reply lib/bootstrap.js]:
```js
arg !== command.name && arg !== command.alias
```
The respective statement appear to be redundant atleast IMO.

Comment (jamesgeorge007) [reply lib/bootstrap.js]:
```js
process.argv.includes(cmd.name) || process.argv.includes(cmd.alias)
```
same here :thinking:

File: lib/bootstrap.js (hunk 8 above)
Comment (jamesgeorge007) [reply lib/bootstrap.js]:
```suggestion
const isHelpFlagPresent = args => args.find(arg => ['help', '--help'].indexOf(arg) !== -1);
```

File: test/binCases/help/mixed/mixed.test.js (hunk 6 above)
Comment (evenstensberg) [reply test/binCases/help/mixed/mixed.test.js]: probably because they tend to get very big

**Comment by evenstensberg** [comment]
What do you propose @ematipico ?

**Comment by evenstensberg** [comment]
We can change the naming later

**Pull request merged**